### PR TITLE
Adding Pushmix Channel

### PR DESCRIPTION
**Pull request by alexpechkarev**

### CODE REVIEW

**Comment by atymic** [comment]
Hi!

Are you still open to creating and maintaining this channel?

If so, could you update it to support laravel 5.5-6.0 please?

Thanks!

**Comment by atymic** [comment]
Hey @alexpechkarev 

Are you still planning on updating this channel?

**Comment by alexpechkarev** [comment]
Hi @atymic

Yes, still interesting in updating the channel.

**Comment by atymic** [comment]
@alexpechkarev have you had a chance to look at the channel?

**Comment by alexpechkarev** [comment]
yes, the changes for Laravel 6 support have been applied.

**Comment by atymic** [comment]
Oh, my bad. Will review now :)

File: README.md [hunk 1]
```diff
@@ -1,5 +1,149 @@
-# new-channels
+# Pushmix notifications channel for Laravel 5.3+
```
Comment (atymic) [inline README.md]:
```suggestion
# Pushmix notifications channel for Laravel 5.5+ & 6.x
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,149 @@
...
 
-Discuss about new channel proposals our share your finished channels in the [proposal issue](https://github.com/laravel-notification-channels/new-channels/issues/6).
+This package makes it easy to send [Pushmix notifications](https://www.pushmix.co.uk/docs/laravel-notification-channel) with Laravel 5.3+.
```
Comment (atymic) [inline README.md]:
```suggestion
This package makes it easy to send [Pushmix notifications](https://www.pushmix.co.uk/docs/laravel-notification-channel) with Laravel 5.5+ & 6.x.
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,149 @@
...
+
+## Installation
+
+You can install the package via composer:
+
+```bash
+$ composer require laravel-notification-channels/pushmix
+```
+
+If you're installing the package in Laravel 5.4 or lower, you must import the service provider:
```
Comment (atymic) [inline README.md]: remove this, since < 5.4 is not supported

File: build/coverage.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
+
+Code Coverage Report:
```
Comment (atymic) [inline build/coverage.txt]: this dir should be gitignored

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
+{
+    "name": "alexpechkarev/new-channels",
```
Comment (atymic) [inline composer.json]: correct this please

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            "name": "Alexander Pechkarev",
+            "email": "alexpechkarev@gmail.com",
+            "homepage": "https://www.pushmix.co.uk",
+            "role": "Developer"
+        }
+    ],
+    "require": {
+        "php": ">=7.2.0",
+        "illuminate/notifications": "6.*",
```
Comment (atymic) [inline composer.json]:
```suggestion
        "illuminate/notifications": "~5.5 | ~6.0",
```

File: composer.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        "illuminate/support": "6.*"
```
Comment (atymic) [inline composer.json]:
```suggestion
        "illuminate/support": "~5.5 | ~6.0",
```

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "require-dev": {
+      "mockery/mockery": "^1.0",
+      "orchestra/testbench": "^3.5",
```
Comment (atymic) [inline composer.json]:
```suggestion
      "orchestra/testbench": "~3.5 || ~4.0",
```

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      "phpunit/phpunit": "^6.4"
```
Comment (atymic) [inline composer.json]:
```suggestion
      "phpunit/phpunit": "^8.0"
```

File: composer.lock [hunk 10]
```diff
@@ -0,0 +1,3685 @@
+{
+    "_readme": [
```
Comment (atymic) [inline composer.lock]: don't commit this for packages

File: phpunit.xml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    </testsuites>
+    <filter>
+        <whitelist>
+            <directory suffix=".php">src/</directory>
+        </whitelist>
+    </filter>
+    <logging>
+        <log type="tap" target="build/report.tap"/>
+        <log type="junit" target="build/report.junit.xml"/>
+        <log type="coverage-html" target="build/coverage" charset="UTF-8" yui="true" highlight="true"/>
```
Comment (atymic) [inline phpunit.xml]:
```suggestion
        <log type="coverage-html" target="build/coverage"/>
```

File: phpunit.xml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+         backupStaticAttributes="false"
+         colors="true"
+         verbose="true"
+         convertErrorsToExceptions="true"
+         convertNoticesToExceptions="true"
+         convertWarningsToExceptions="true"
+         processIsolation="false"
+         stopOnFailure="false">
+    <testsuites>
+        <testsuite name="Spatie Test Suite">
```
Comment (atymic) [inline phpunit.xml]: Copy/paste?

File: src/PushmixChannel.php [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+use NotificationChannels\Pushmix\Exceptions\CouldNotSendNotification;
+
+class PushmixChannel
+{
+    /** @var PushmixClient */
+    protected $pusmixClient;
+
+    public function __construct(PushmixClient $pusmixClient)
+    {
+        $this->pusmixClient = $pusmixClient;
```
Comment (atymic) [inline src/PushmixChannel.php]: `pusmixClient` should be `pushmixClient`

File: src/PushmixChannel.php [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param mixed $notifiable
+     * @param \Illuminate\Notifications\Notification $notification
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     * @throws \NotificationChannels\Pushmix\Exceptions\CouldNotSendNotification
+     */
+    public function send($notifiable, Notification $notification)
+    {
+        if (! $to = $notifiable->routeNotificationFor('Pushmix')) {
```
Comment (atymic) [inline src/PushmixChannel.php]:
```suggestion
        if (! $to = $notifiable->routeNotificationFor('Pushmix')) {
```

lowercase for routing

File: src/PushmixChannel.php [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+        // Call with parameters
+        try {
+            return $this->pusmixClient->sendNotification($parameters);
+        } catch (RequestException $e) {
+            throw CouldNotSendNotification::error($e);
+        }
+    }
+
+    /***/
```
Comment (atymic) [inline src/PushmixChannel.php]:
```suggestion
```

File: src/PushmixClient.php [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+namespace NotificationChannels\Pushmix;
+
+use GuzzleHttp\Client;
+use NotificationChannels\Pushmix\Exceptions\InvalidConfiguration;
+
+class PushmixClient
+{
+    protected $api_url = 'https://www.pushmix.co.uk/api/notify';
+
+    protected $client;
```
Comment (atymic) [inline src/PushmixClient.php]: docblocks please

File: src/PushmixClient.php [hunk 17]
```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace NotificationChannels\Pushmix;
+
+use GuzzleHttp\Client;
+use NotificationChannels\Pushmix\Exceptions\InvalidConfiguration;
+
+class PushmixClient
+{
+    protected $api_url = 'https://www.pushmix.co.uk/api/notify';
```
Comment (atymic) [inline src/PushmixClient.php]: this can be a const

File: src/PushmixClient.php [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+    /**
+     * Initialize additional parameters.
+     */
+    public function initKey()
+    {
+        if (is_null(config('services.pushmix.key', null))) {
+            throw InvalidConfiguration::configurationNotSet();
+        }
+
+        $this->additionalParams = [
```
Comment (atymic) [inline src/PushmixClient.php]: formatting is wonky here

File: src/PushmixClient.php [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+    /**
+     * Get requestAsync.
+     * @return bool
+     */
+    public function getRequestAsync()
+    {
+        return $this->requestAsync;
+    }
+
+    /***/
```
Comment (atymic) [inline src/PushmixClient.php]: Can you remove all these?

File: src/PushmixClient.php [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+    /**
+     * Merge Notification Parameters and Send Notification to Pushmix API.
+     *
+     * @param array $parameters notification parameters
+     * @return mixed
+     */
+    public function sendNotification($parameters)
+    {
+        $parameters = array_merge($parameters, $this->additionalParams);
+        $this->headers['body'] = json_encode($parameters);
```
Comment (atymic) [inline src/PushmixClient.php]: `$this->headers` should be called something like `$this->requestOptions`, as it doesnt contain headers

File: src/PushmixServiceProvider.php [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+     * Get the services provided by the provider.
+     *
+     * @return array
+     */
+    public function provides()
+    {
+        return ['pushmix'];
+    }
+
+    /***/
```
Comment (atymic) [inline src/PushmixServiceProvider.php]: as with other files, remove this

**Review by atymic — CHANGES_REQUESTED** [review]

**Comment by atymic** [comment]
Have added some review comments, re-request a review once you've looked over/fixed the issues :)

**Comment by atymic** [comment]
Closing due to inactivity, ping me once changes are made.

**Pull request closed without merging**